File: ractor/nginx.rb
PATTERN = /^(?<host>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>(?:[^\"]|\\")*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>(?:[^\"]|\\")*)" "(?<agent>(?:[^\"]|\\")*)")?$/

def parse(line)
  m = PATTERN.match(line)
  return m.named_captures
end

record = nil
$stdin.each_line(chomp: true) do |line|
  record = parse(line)
end

puts(record)

File: ractor/nginx.py
import re
import sys

PATTERN = re.compile(r'^(?P<host>[^ ]*) [^ ]* (?P<user>[^ ]*) \[(?P<time>[^\]]*)\] "(?P<method>\S+)(?: +(?P<path>(?:[^\"]|\\")*?)(?: +\S*)?)?" (?P<code>[^ ]*) (?P<size>[^ ]*)(?: "(?P<referer>(?:[^\"]|\\")*)" "(?P<agent>(?:[^\"]|\\")*)")?$')

def parse(line):
  m = PATTERN.match(line)
  return m.groupdict()

for line in sys.stdin:
  record = parse(line.strip())

print(record)
